spec/index.bs

```diff
@@ -1403,6 +1427,97 @@ but which the user agent has not been updated to recognize yet. If the [=set ent
+        hardware like a serial number.
+
+    : <dfn>description</dfn>
+    ::
+        A human readable string describing the [=adapter=] as reported by the driver, if avaliable.
+        Empty string otherwise.
+
+        Note: Because no formatting is applied to {{GPUAdapterInfo/description}} attempting to parse
+        this value is not recommended. Applications which change their behavior based on the
+        {{GPUAdapterInfo}}, such as applying workarounds for known driver issues, should rely on the
+        other fields when possible.
+
+</dl>
+
+<div algorithm>
+    To create a <dfn abstract-op>new adapter info</dfn> for a given [=adapter=] |adapter|, run the
+    following steps:
+
+    1. Let |adapterInfo| be a new {{GPUAdapterInfo}}.
+    1. Let |unmaskedValues| be |adapter|.{{adapter/[[unmaskedIdentifiers]]}}
+    1. If |unmaskedValues| [=set/contains=] `"vendor"` and the vendor is known:
+        1. Set |adapterInfo|.{{GPUAdapterInfo/vendor}} to the name of |adapter|'s vendor as a
+            [$normalized identifier string$].
+        1. Otherwise set |adapterInfo|.{{GPUAdapterInfo/vendor}} to the empty string or a
+            reasonable approximation of the vendor as a [$normalized identifier string$].
+    1. If |unmaskedValues| [=set/contains=] `"architecture"` and the architecture is known:
+        1. Set |adapterInfo|.{{GPUAdapterInfo/architecture}} to a [$normalized identifier string$]
+            representing the family or class of adapters to which |adapter| belongs.
+        1. Otherwise set |adapterInfo|.{{GPUAdapterInfo/architecture}} to the empty string or a
+            reasonable approximation of the architecture as a [$normalized identifier string$].
+    1. If |unmaskedValues| [=set/contains=] `"deviceId"` and the deviceId is known:
+        1. Set |adapterInfo|.{{GPUAdapterInfo/deviceId}} to a vendor-specific numeric ID for
+            |adapter|.
+        1. Otherwise set |adapterInfo|.{{GPUAdapterInfo/deviceId}} to `0` or a reasonable
+            approximation of the vendor-specific numeric ID for |adapter|.
+    1. If |unmaskedValues| [=set/contains=] `"description"` and a description is known:
+        1. Set |adapterInfo|.{{GPUAdapterInfo/description}} to a description of the |adapter| as
+            reported by the driver.
+        1. Otherwise set |adapterInfo|.{{GPUAdapterInfo/vendor}} to the empty string or a
+            reasonable approximation of a description.
+    1. Return |adapterInfo|.
+</div>
+
+<div algorithm>
+    To generate a <dfn abstract-op>normalized identifier string</dfn> for a given {{DOMString}}
+    |identifier|, run the following steps:
+
+    1. Let |normalized| be a copy of |identifier|.
+    1. Remove any non-ASCII characters from |normalized|.
+    1. Convert |normalized| to lowercase.
+    1. Replace any spaces in |normalized| with a hyphen character. (`"-"`)
+    1. Return |normalized|.
+</div>
```

Comment:
nit: Since we don't actually apply this as an algorithm anywhere, how about describing this as a set of rules instead? Basically it must match `[a-z0-9]([a-z0-9-]*[a-z0-9])?`.

Reply:
This algorithm is applied twice in the `new adapter info` section.

Reply:
It's referenced, but IIUC isn't actually called with any specific inputs. It sort of alludes to being called with values from the driver, but since it's not concrete I think we could basically just say the UA comes up with some string matching that regex, however it wants to.